### PR TITLE
Add SQLExecuteQueryOperator operator

### PR DESCRIPTION
**Pull request by mik-laj**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by zikun** [comment]
Nice consolidation! I was also thinking about the same thing.

On the naming, what do you think about just `SQLOperator`?
We have `BranchPythonOperator` -> `PythonOperator` (We did not call it `PythonRunCodeOperator`), so `BranchSQLOperator` -> `SQLOperator`

File: airflow/contrib/operators/vertica_operator.py [hunk 1]
```diff
@@ -15,14 +15,35 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-"""This module is deprecated. Please use `airflow.providers.vertica.operators.vertica`."""
+"""This module is deprecated. Please use `airflow.providers.sqlite.operators.sqlite`."""
```
Comment (turbaszek) [inline airflow/contrib/operators/vertica_operator.py:18]: sqlite? 
```suggestion
"""This module is deprecated. Please use `airflow.providers.sql.operators.sql`."""
```

File: tests/operators/test_sql.py [hunk 2]
```diff
@@ -742,3 +742,68 @@ def test_with_skip_in_branch_downstream_dependencies2(self, mock_hook):
...
+            dummy VARCHAR(50)
+        );
+        """
+        op = SQLExecuteQueryOperator(
+            task_id='basic_sqlite',
+            sql=sql,
+            conn_id='sqlite_default',
+            dag=self.dag,
+        )
+        op.run(start_date=DEFAULT_DATE, end_date=DEFAULT_DATE, ignore_ti_state=True)
```
Comment (turbaszek) [inline tests/operators/test_sql.py:777]: Does this create side effect on db level? Should we clear database after tests?

**Comment by mik-laj** [comment]
> On the naming, what do you think about just SQLOperator?

Now we are trying to give more ambiguous names to make it easier to add new operators and to understand the operator's behavior without reading its description. 
A good example is BigQueryOperator, which could do multiple operations, but In Airflow 2.0, we've given it the more precisive name BigQueryExecuteQuery.  This allows this operator to be easily distinguished from other operators that perform other operations.
More info:
https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#naming-conventions-for-provider-packages

**Comment by potiuk** [comment]
Hey @mik-laj -> is it something we will managed add before 2.0.0rc1? If not, can you please change the milestone?

**Comment by mik-laj** [comment]
It is unlikely that I will find time for this.

**Comment by eladkal** [comment]
closed in favor of https://github.com/apache/airflow/pull/25717

**Pull request closed without merging**